setup.py

```diff
@@ -21,53 +21,13 @@
 
 PYTHON_BINDINGS_PATH = os.path.dirname(os.path.abspath(__file__))
 
-class MpiImplementations(Enum):
-    OPENMPI = 1
-    MPICH = 2
-
-
-def check_mpi_implementation(mpi_compiler_wrapper):
-    FNULL = open(os.devnull, 'w')  # used to supress output of subprocess.call
-
-    if subprocess.call([mpi_compiler_wrapper, "-showme:compile"], stdout=FNULL, stderr=FNULL) == 0:
-        PRECICE_MPI_IMPLEMENTATION = MpiImplementations.OPENMPI
-    elif subprocess.call([mpi_compiler_wrapper, "-compile-info"], stdout=FNULL, stderr=FNULL) == 0:
-        PRECICE_MPI_IMPLEMENTATION = MpiImplementations.MPICH
-    else:
-        raise Exception("unknown/no mpi++")
-
-    return PRECICE_MPI_IMPLEMENTATION
-
-
-def determine_mpi_args(mpi_compiler_wrapper):
-    PRECICE_MPI_IMPLEMENTATION = check_mpi_implementation(mpi_compiler_wrapper)
-    # determine which flags to use with mpi compiler wrapper
-    if PRECICE_MPI_IMPLEMENTATION is MpiImplementations.OPENMPI:
-        mpi_compile_args = subprocess.check_output([mpi_compiler_wrapper, "-showme:compile"]).decode().strip().split(
-            ' ')
-        mpi_link_args = subprocess.check_output([mpi_compiler_wrapper, "-showme:link"]).decode().strip().split(' ')
-    elif PRECICE_MPI_IMPLEMENTATION is MpiImplementations.MPICH:
-        mpi_compile_args = subprocess.check_output([mpi_compiler_wrapper, "-compile-info"]).decode().strip().split(' ')[
-                           1::]
-        mpi_link_args = subprocess.check_output([mpi_compiler_wrapper, "-link-info"]).decode().strip().split(' ')[1::]
-    else:  # if PRECICE_MPI_IMPLEMENTATION is not mpich or openmpi quit.
-        raise Exception("unknown/no mpi found using compiler %s. Could not build PySolverInterface." % mpi_compiler_wrapper)
-
-    return mpi_compile_args, mpi_link_args
-
-
-def get_extensions(mpi_compiler_wrapper, is_test):
+def get_extensions(is_test):
     compile_args = []
     link_args = []
```

Comment:
This list can now contain only a single argument. Maybe there is a more elegant way of writing this?

Reply:
The one-element-list is indeed a bit strange, but I currently do not really see a nicer solution. Note that we also want to be able to provide an empty list, if we are running the tests. I would keep it as it is.